pkg/api/pullrequestpayloadtest/v1/types.go

```diff
@@ -0,0 +1,48 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PullRequestPayloadTest represents the intent to run a battery of OCP release
+// payload validating jobs
+type PullRequestPayloadTest struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   PullRequestPayloadTestSpec   `json:"spec"`
+	Status PullRequestPayloadTestStatus `json:"status"`
+}
+
+type ReleaseJobSpec struct {
+	Org     string `json:"org"`
+	Repo    string `json:"repo"`
+	Branch  string `json:"branch"`
+	Variant string `json:"variant,omitempty"`
```

Comment:
An optional coordinate seems odd.  What is a variant?

Reply:
The coordinates address a test in a specific file and variant is an optional coordinate:

See
 https://github.com/openshift/release/blob/master/ci-operator/config/openshift/release/openshift-release-master__ci-4.9.yaml#L301-L305 and compare to e.g.
https://github.com/openshift/release/blob/master/ci-operator/config/openshift/metallb-operator/openshift-metallb-operator-main.yaml#L67-L70

Also: https://docs.ci.openshift.org/docs/how-tos/contributing-openshift-release/#variants

I think that at the moment, release jobs happen to *always* be present in a variant file, but that's just an arbitrary state, not something to rely on.